src/jvm/test/resources/expected_logs/run_concurrent_test/main_thread_blocked_trace_debugger.txt

```diff
@@ -5,10 +5,9 @@ The following interleaving leads to the error:
 |                                                                    Main Thread                                                                     |
 | -------------------------------------------------------------------------------------------------------------------------------------------------- |
 | $this_runCatching.block() at BaseRunConcurrentRepresentationTest.testRunWithModelChecker$lambda$1$lambda$0(RunConcurrentRepresentationTests.kt:47) |
-|   block() at MainThreadBlockedRunConcurrentRepresentationTest.block(RunConcurrentRepresentationTests.kt:96)                                        |
-|     q.take() at MainThreadBlockedRunConcurrentRepresentationTest.block(RunConcurrentRepresentationTests.kt:101)                                    |
+|   block() at MainThreadBlockedRunConcurrentRepresentationTest.block(RunConcurrentRepresentationTests.kt:98)                                        |
+|     q.take() at MainThreadBlockedRunConcurrentRepresentationTest.block(RunConcurrentRepresentationTests.kt:103)                                    |
 |       lock.lockInterruptibly() at ArrayBlockingQueue.take(ArrayBlockingQueue.java:417)                                                             |
-|       count ➜ 0 at ArrayBlockingQueue.take(ArrayBlockingQueue.java:419)                                                                            |
```

Comment:
BTW, why these `count` accesses disappeared from the trace

Reply:
`q.take()` is a function operated on a collection. According to our new hiding rules it shows the function (just like `add` and `put` etc..)

Now this function has three child nodes: `lock.lockInterruptibly()`, `count -> 0`, `notEmpty.await()`.

The first and last function call are not part of the collection stdLib and are therefore not hidden. However, the count -> 0 event is part of the library and therefore is hidden.

Reply:
Shouldn't Lincheck hide either everything or nothing in this case? What is the expected semantics here?

Reply:
That wouldn't work for functions taking lambda's. For instance, if we would hide everything from `computeIfAbsent` we would not see user code. But keeping everything would leave a lot of noise. 

Therefore we cannot blindly hide everything inside a function and the resulting semantics are that we only hide what is known that should be hidden (in this PR the std Collections).

Similarly, when a switchpoint does occus in one of these collections, we do not want to show everything that happened inside. Only the calls leading up to the switch event.

Reply:
The question is mainly about the resulting trace we wish to show here, not the implementation. What would you, as a user, prefer to see, and how can we get there?

Reply:
We could also hide all contents until we find a lambda or runnable, but a lambda can also be part of some internal logic..